### ISSUE TITLE
vicadmin server consumes 100% CPU in 0.5 tag candidate

### ISSUE THREAD

**Issue by corrieb** [vmware/vic#1991]
I have tried to reproduce the steps I went through initially, but have been unable to. However, I have a log bundle with pprof enabled which I will attach here. Logging into the appliance confirms with top that it's the admin server looping, not being restarted.

I will keep the VCH alive as long as I can so that it can be poked if necessary.

[logs.zip](https://github.com/vmware/vic/files/419166/logs.zip)

**Comment by sgairo** [comment]
@corrieb were you running multiple VCH off of the same host?

**Comment by corrieb** [comment]
This is against VC. Yes I think there was another VCH on there at the time

**Comment by hickeng** [comment]
I grabbed a profile via :6061/debug/pprof/profile (which we really should include in the bundles!).
[vic-admin.zip](https://github.com/vmware/vic/files/419188/vic-admin.zip)

The `top` from profile is:
```
vagrant@devbox:~$ go tool pprof ./vicadmin ./vic-admin.profile
Entering interactive mode (type "help" for commands)
(pprof) top
29.59s of 29.96s total (98.77%)
Dropped 10 nodes (cum <= 0.15s)
Showing top 10 nodes out of 15 (cum >= 0.16s)
      flat  flat%   sum%        cum   cum%
    22.74s 75.90% 75.90%     22.74s 75.90%  bytes.LastIndex
     4.77s 15.92% 91.82%      6.62s 22.10%  syscall.Syscall6
     0.66s  2.20% 94.03%      0.66s  2.20%  runtime/internal/atomic.Cas
     0.32s  1.07% 95.09%      0.83s  2.77%  runtime.reentersyscall
     0.22s  0.73% 95.83%      0.66s  2.20%  runtime.casgstatus
     0.20s  0.67% 96.50%     29.94s 99.93%  main.findSeekPos
     0.18s   0.6% 97.10%      0.91s  3.04%  runtime.exitsyscall
     0.18s   0.6% 97.70%      0.40s  1.34%  runtime.exitsyscallfast
     0.16s  0.53% 98.23%      6.89s 23.00%  os.(*File).ReadAt
     0.16s  0.53% 98.77%      0.16s  0.53%  runtime/internal/atomic.Store
(pprof)
```

and cross-referencing that with the stack traces from the bundle then this thread is maxing CPU:
```
goroutine 23 [runnable]:
os.(*File).ReadAt(0xc82001c030, 0xc82050d880, 0x400, 0x400, 0x33d2, 0x1, 0xffffffffffffffff, 0x0)
	/usr/local/go/src/os/file.go:109
main.findSeekPos(0xc82001c030, 0x0)
	/drone/src/github.com/vmware/vic/cmd/vicadmin/logs.go:383 +0x207
main.tailFile(0x7f5e940e0960, 0xc820bbe180, 0xc82023bb30, 0x23, 0xc82001c028, 0x0, 0x0)
	/drone/src/github.com/vmware/vic/cmd/vicadmin/logs.go:322 +0x186
created by main.(*server).tailFiles
	/drone/src/github.com/vmware/vic/cmd/vicadmin/server.go:246 +0x385
```

**Comment by sgairo** [comment]
what was the space allotted to your host? I have had a similar issue in the past and it was because I was creating multiple VCH against VC with the same host but there wasn't enough space. Once I increased it, it works fine.

**Comment by corrieb** [comment]
@sgairo plenty of space. < 40% memory used. Plenty of CPU and disk

**Comment by hickeng** [comment]
This is almost certainly a code problem in `main.findSeekPos` - I'm not sure precisely what. I *highly* recommend some unit tests to track down this problem once a hypothesis as to the exact issue is formed.

**Comment by corrieb** [comment]
Yea looking at it, this is exactly the kind of code that's very hard to verify visually

**Comment by kjplatz** [comment]
Fixed by PR #2035  and unit tests implemented
